### PR TITLE
feat(content): thermal electric survival outfit

### PR DESCRIPTION
**Pull request by yay855**
Added thermal electric survival outfit, a temperature-regulating piece of clothing, plus recipe.

### Required

- [x] I wrote the PR title in [conventional commit format](https://docs.cataclysmbn.org/en/contribute/changelog_guidelines/).
- [x] I ran the [code formatter](https://docs.cataclysmbn.org/en/contribute/contributing/#code-style).
- [ ] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [x] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

## Purpose of change

Survival in summer is not difficult but _tedious_ due to most clothing not having temperature regulation for high temperatures, and armors having relatively high thermal insulation. It means survivors can't really fight during the summer.

## Describe the solution

Added thermal electric survival outfit, a piece of clothing that regulates the wearer's temperature when turned on. Also added crafting recipe.

## Describe alternatives you've considered

Overhauling the way clothing insulation works to have a maximum and minimum insulation.

## Testing

It still needs testing and possibly loot table additions, but the basic item and recipe are completed for now.

### CODE REVIEW

**Comment by Soadreqm** [comment]
I'd suggest fluffing it as a [liquid cooling garment](https://en.wikipedia.org/wiki/Liquid_cooling_and_ventilation_garment) instead. Just replace some of the heating elements used to craft it with rubber hoses. Leave a few to explain how it can also heat the liquid. Maybe add it to firefighter equipment spawns.

**Comment by Proxiehunter** [comment]
> I'd suggest fluffing it as a [liquid cooling garment](https://en.wikipedia.org/wiki/Liquid_cooling_and_ventilation_garment) instead. Just replace some of the heating elements used to craft it with rubber hoses. Leave a few to explain how it can also heat the liquid. Maybe add it to firefighter equipment spawns.

If you go that route I don't see a reason to not also allow _leather_ hoses not just rubber ones.

File: data/json/items/tool_armor.json [hunk 1]
```diff
@@ -619,6 +619,34 @@
     "warmth": 60,
     "magazine_well": "250 ml"
   },
+  {
+    "id": "thermal_super",
+    "copy-from": "thermal_outfit",
+    "name": { "str": "thermal electric survival outfit" },
+    "description": "This skintight suit covers you from head to toe and is designed to keep you at an ideal temperature using a number of heatsinks and heating coils.  It also includes some handy features such as a built-in thermometer.  Use it to turn it on.",
+    "flags": [ "VARSIZE", "SKINTIGHT", "TRADER_AVOID", "ONLY_ONE", "ALARMCLOCK", "THERMOMETER" ],
```
Comment (chaosvolt) [inline data/json/items/tool_armor.json:627]:
```suggestion
    "flags": [ "VARSIZE", "SKINTIGHT", "TRADER_AVOID", "ONLY_ONE", "THERMOMETER" ],
```

Has `ALARMCLOCK` flag but recipe only adds a thermometer.

Comment (yay855) [reply data/json/items/tool_armor.json]: I was debating adding a watch to the recipe and making it a clock, but I decided not to- and apparently forgot to remove the flag.

**Review by chaosvolt — APPROVED** [review]

**Comment by ZChris13** [comment]
this is cool

**Pull request merged**